equinox/nn/_embedding.py

```diff
@@ -194,9 +197,14 @@ def precompute_freqs_cis(
         freqs = 1.0 / (
             theta ** (jnp.arange(0, embedding_size, 2)[jnp.newaxis, :] / embedding_size)
         )
-        t = jnp.arange(end)
+
+        t = jnp.arange(end, dtype=float)
```

Comment:
Heads-up that using `dtype=int` (and so perhaps also `dtype=float`?) can cause funny platform-dependent behaviour -- see https://github.com/google/jax/issues/9574.

If you want "default floating point precision" here then I think `jnp.arange(float(end))` will accomplish that: `jnp.arange` will check whether its argument is exact or inexact, and should then use the appropriate default dtype of that sort.

Reply:
Fixed this for the sinusoidal embeddings, too. There, it was enough to make one of the arguments a float, which makes the whole thing a float, i.e.:
```python
jax.numpy.arange(0.0, 42, 2)
```